frontend/src/components/data-table/column-header.tsx

```diff
@@ -137,6 +113,37 @@ export const DataTableColumnHeader = <TData, TValue>({
     );
   };
 
+  const dtype: DataType = mapDTypeToDataType(column.columnDef.meta?.dtype);
+  const columnFormatOptions = dtype ? formatOptions[dtype] : [];
+
+  const renderFormatOptions = () => {
+    if (
+      columnFormatOptions.length === 0
+      // ||
+      // !column.getCanFormat?.() ||
+      // !column.getColumnFormatting
+    ) {
+      return null;
+    }
+    return (
+      <DropdownMenuSub>
+        <DropdownMenuSubTrigger>
+          <WrapTextIcon className="mr-2 h-3.5 w-3.5 text-muted-foreground/70" />
+          Format
+        </DropdownMenuSubTrigger>
+        <DropdownMenuPortal>
+          <DropdownMenuSubContent>
+            {columnFormatOptions.map((option) => (
+              <DropdownMenuItem key={option} onClick={() => setFormat(option)}>
```

Comment:
this can be `onClick={() => column. setColumnFormatting(option)}>`

we should also add:
```
{Boolean(column.getColumnFormatting()) && 
  <DropdownMenuItem key={option} onClick={() => column.setColumnFormatting(undefined)}>
}
```